token/cli/src/config.rs

```diff
@@ -29,53 +37,64 @@ pub(crate) struct Config<'a> {
     pub(crate) default_keypair: KeypairOrPath,
     pub(crate) nonce_account: Option<Pubkey>,
     pub(crate) nonce_authority: Option<Pubkey>,
-    pub(crate) blockhash_query: BlockhashQuery,
     pub(crate) sign_only: bool,
     pub(crate) dump_transaction_message: bool,
     pub(crate) multisigner_pubkeys: Vec<&'a Pubkey>,
-    pub(crate) program_id: Pubkey,
```

Comment:
I hoped to do it that way at first, but since we want the CLI to figure out the program id, a non-optional program id makes that much less usable.  With this change, the parameter is only used for offline signing and creating new mints, otherwise the CLI always figures it out.